pkg/operator/ceph/csi/cluster_config.go

```diff
@@ -278,3 +282,75 @@ func SaveClusterConfig(clientset kubernetes.Interface, clusterNamespace string,
 
 	return nil
 }
+
+// updateCSIDriverOptions updates the CSI driver options, including read affinity, kernel mount options
+// and fuse mount options, for all entries belonging to the same cluster.
+func updateCSIDriverOptions(curr, clusterKey string,
+	csiDriverOptions *cephv1.CSIDriverSpec) (string, error) {
+	cc, err := parseCsiClusterConfig(curr)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse current csi cluster config")
+	}
+
+	for i := range cc {
+		// If the clusterID belongs to the same cluster, update the entry.
+		if clusterKey == cc[i].Namespace {
+			cc[i].ReadAffinity = &csiDriverOptions.ReadAffinity
+			if cc[i].CephFS == nil {
+				cc[i].CephFS = &CsiCephFSSpec{
+					KernelMountOptions: csiDriverOptions.CephFS.KernelMountOptions,
+					FuseMountOptions:   csiDriverOptions.CephFS.FuseMountOptions,
+				}
+			} else {
+				cc[i].CephFS.KernelMountOptions = csiDriverOptions.CephFS.KernelMountOptions
+				cc[i].CephFS.FuseMountOptions = csiDriverOptions.CephFS.FuseMountOptions
+			}
+		}
+	}
+
+	return formatCsiClusterConfig(cc)
+}
+
+// SaveCSIDriverOptions, similar to SaveClusterConfig, updates the config map used by ceph-csi
+// with CSI driver options such as read affinity, kernel mount options and fuse mount options.
+func SaveCSIDriverOptions(clientset kubernetes.Interface, clusterNamespace string, clusterInfo *cephclient.ClusterInfo) error {
+	// csi is deployed into the same namespace as the operator
+	csiNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
+	if csiNamespace == "" {
+		logger.Warningf("cannot save csi config due to missing env var %q", k8sutil.PodNamespaceEnvVar)
+		return nil
+	}
+	logger.Debugf("using %q for csi configmap namespace", csiNamespace)
+
+	configMutex.Lock()
+	defer configMutex.Unlock()
+
+	// fetch current ConfigMap contents
+	configMap, err := clientset.CoreV1().ConfigMaps(csiNamespace).Get(clusterInfo.Context, ConfigName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch current csi config map")
+	}
+
+	// update ConfigMap contents for current cluster
+	currData := configMap.Data[ConfigKey]
+	if currData == "" {
+		currData = "[]"
+	}
+
+	newData, err := updateCSIDriverOptions(currData, clusterNamespace, &clusterInfo.CSIDriverSpec)
```

Comment:
Does ceph CSI  already have the changes to handle the new config?

Reply:
> Does ceph CSI already have the changes to handle the new config?

nope, it will come with release 3.10 but this is a backward compatible non-breaking change so it should be fine.